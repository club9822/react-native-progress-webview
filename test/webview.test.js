import React from 'react';
import { shallow } from 'enzyme';
import WebView from '../lib/webview';

describe('WebView', () => {
	describe('Rendering', () => {
		it('should match to snapshot', () => {
			const component = shallow(<WebView source={{ uri: 'https://www.google.com.au' }}/>);
			expect(component).toMatchSnapshot()
		});
	});

	describe('function callback', () => {
		it('should call onLoadProgress', () => {
			const onLoadProgress = jest.fn();
			shallow(
				<WebView
					source={{ uri: 'https://www.google.com.au' }}
					onLoadProgress={onLoadProgress}
				/>
			);

			expect(onLoadProgress).toBeCalled();
		});
	});
});
