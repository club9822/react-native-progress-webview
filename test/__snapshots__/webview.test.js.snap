// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WebView Rendering should match to snapshot 1`] = `ShallowWrapper {}`;
